### PR TITLE
Close #183: Add assertions and logging

### LINKED ISSUE #183: Defensive coding: Azeem

### PR DESCRIPTION
**Pull request by DrWala**
### What this does
- Closes #183 by adding assertions to document assumptions, and adding logging to assist in tracing

### How to test
- Checkout the PR and run tests, ensure all tests pass.

### CODE REVIEW

File: src/main/java/seedu/address/logic/commands/room/ListRoomCommand.java [hunk 1]
```diff
@@ -15,12 +18,14 @@
     public static final String COMMAND_WORD = "olist";
 
     public static final String MESSAGE_SUCCESS = "Listed all rooms";
+    private static final Logger logger = LogsCenter.getLogger(AddRoomCommand.class);
```
Comment (cnlinh) [inline src/main/java/seedu/address/logic/commands/room/ListRoomCommand.java:21]: Wrong class name?

**Review by cnlinh — CHANGES_REQUESTED** [review]
Appropriate use of assertion and logging, just one minor edit

File: src/main/java/seedu/address/logic/commands/room/ListRoomCommand.java (hunk 1 above)
Comment (DrWala) [reply src/main/java/seedu/address/logic/commands/room/ListRoomCommand.java:21]: Good catch, thanks.

**Review by cnlinh — APPROVED** [review]
LGTM

**Pull request merged**